doc/Syscalls.md

```diff
@@ -224,7 +225,8 @@ additional meaning such as the number of devices present, as is the case in the
 ### 3: Allow
 
 Allow marks a region of memory as shared between the kernel and application.
-A null pointer revokes sharing a region.
+Passing a null pointer requests the corresponding driver to stop accessing the
+shared memory region.
```

Comment:
I _think_ the intent was actually a strong revoke -- namely that if a capsule held an `AppSlice` it would fail on the next attempt to access that `AppSlice`. I'm not sure that's implemented, but I think that's still the aspiration.

We could consider updating this to caveat that this is imperfectly implemented, but I don't think we want to change this contract?

Reply:
I think it should be as strong as the subscribe case. That is, there is expected behavior that the appslice will not still be used, but if the system doesn't behave that way then the fault lies with the capsule and not with the kernel.

Reply:
I've observed some disagreement between contributors on what the contract is. I want to resolve that by making the documentation reflect the current implementation (note that this is an important part of the Tock threat model).

I prefer to have a strong revoke, but that's a chance that will require a significant amount of engineering effort to implement.